File: src/main/kotlin/application/controller/dialogs/rule/misc/MargolusDialog.java
package application.controller.dialogs.rule.misc;

import application.controller.dialogs.rule.RuleWidget;
import application.model.rules.misc.Margolus;

public class MargolusDialog extends RuleWidget {
    public MargolusDialog() {
        super();

        ruleFamily = new Margolus();
    }

    @Override
    public void updateRule(String rulestring) {
        ruleFamily.setRulestring(rulestring);
    }
}

File: src/main/kotlin/application/controller/dialogs/rule/misc/MargolusDialog.kt
package application.controller.dialogs.rule.misc

import application.controller.dialogs.rule.RuleWidget
import application.model.rules.misc.Margolus

class MargolusDialog : RuleWidget() {
    override fun updateRule(rulestring: String?) {
        ruleFamily!!.rulestring = rulestring
    }

    init {
        ruleFamily = Margolus()
    }
}